drivers/gpu/msm/adreno_ringbuffer.c

```diff
@@ -937,14 +973,16 @@ int adreno_ringbuffer_submitcmd(struct adreno_device *adreno_dev,
 		!adreno_is_a3xx(adreno_dev) &&
 		(cmdobj->profiling_buf_entry != NULL)) {
 		user_profiling = true;
-		dwords += 6;
 
 		/*
-		 * REG_TO_MEM packet on A5xx and above needs another ordinal.
-		 * Add 2 more dwords since we do profiling before and after.
+		 * User side profiling uses two IB1s, one before with 4 dwords
+		 * per INDIRECT_BUFFER_PFE call
 		 */
-		if (!ADRENO_LEGACY_PM4(adreno_dev))
-			dwords += 2;
+		// FIXME: Felix: Removing this seems like it'll give legacy too many dwords...
```

Comment:
Could use some guidance here?

Reply:
Check 4.14... That's a nice find btw...

Reply:
```
/* Return true if the hardware uses the legacy (A4XX and older) PM4 format */
#define ADRENO_LEGACY_PM4(_d) (ADRENO_GPUREV(_d) < 500)
```
[adreno_pm4types.h](https://github.com/sonyxperiadev/kernel/blob/aosp/LA.UM.7.1.r1/drivers/gpu/msm_legacy/adreno_pm4types.h#L247-L248)
Guess we don't need to care about a4xx?

Will align with 4.14 then (for reference: https://github.com/sonyxperiadev/kernel/blob/aosp/LA.UM.7.1.r1/drivers/gpu/msm/adreno_ringbuffer.c#L1010)